packages/core/services/reflector.service.ts

```diff
@@ -16,13 +16,21 @@ export interface CreateDecoratorOptions<T = any> {
    * The transform function to apply to the metadata value.
    * @default value => value
    */
-  transform?: (value: T) => T;
+  transform?: (value: TParam) => TTransformed;
 }
 
+export type CreateDecoratorWithTransformOptions<
```

Comment:
```suggestion
type CreateDecoratorWithTransformOptions<
```

I believe we shoulnd't expose this because it's just a helper type (kinda)

Reply:
> I believe we shouldn't expose this because it's just a helper type (kinda)

Agree, just updated.